components/mdns/examples/pytest_mdns.py

```diff
@@ -165,11 +171,25 @@ def test_examples_protocol_mdns(dut):
         ])
         print('Resolving {} using "dig" succeeded with:\n{}'.format(
             specific_host, dig_output))
-        if not ip_address.encode('utf-8') in dig_output:
+        if not ipv4.encode('utf-8') in dig_output:
             raise ValueError(
                 'Test has failed: Incorrectly resolved DUT hostname using dig'
-                "Output should've contained DUT's IP address:{}".format(
-                    ip_address))
+                "Output should've contained DUT's IP address:{}".format(ipv4))
+        # 6. check the DUT reverse lookup
+        if dut.app.sdkconfig.get('MDNS_RESPOND_REVERSE_QUERIES') is True:
+            for ip_address in ip_addresses:
+                dig_output = subprocess.check_output([
+                    'dig', '+short', '-p', '5353', '@224.0.0.251', '-x',
+                    '{}'.format(ip_address)
+                ])
+                print('Reverse lookup for {} using "dig" succeeded with:\n{}'.
+                      format(ip_address, dig_output))
+                if specific_host not in dig_output.decode():
+                    raise ValueError(
+                        'Test has failed: Incorrectly resolved DUT IP address using dig'
+                        "Output should've contained DUT's name:{}".format(
```

Comment:
Is it possible to get also the return here and avoid the print ? Print is useful for debugging, so I think it's better to have it in the Exception message.

Reply:
You mean removing the `dig`'s outputs from the previous print and adding it here to the exception message?

Updated, PTAL. I sill left the `print()` message with basic information (that the lookup succeeded) as I find it useful even if the test passes.

Reply:
I've decided to revert this change, as I find it quite useful to see the actual output of the tool, even in the "happy flow" scenario.
Compare these:
* [prints just info](https://github.com/espressif/esp-protocols/actions/runs/4177245710/jobs/7242835054#step:6:215)
* [prints the result too](https://github.com/espressif/esp-protocols/actions/runs/4127638367/jobs/7131203333#step:6:213)

we're calling it with  `--short` argument, so it prints one more line, which is concise and useful IMO.

Reply:
Yes exactly, I was concerned that the print may get lost in case of error and that it may not be that useful in successful runs.  But as you state it is in fact useful better to have both.